### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by LukaszRozmej**
1. Remove infinite waits -> tests will fail instead of haning
2. Fix infinite loop
3. Fix disposing TestBlockchain
4. Fix ReadOnlyState
5. Make multiple AddFunds to one block

### CODE REVIEW

**Review by marcindsobczak — APPROVED** [review]

File: src/Nethermind/Nethermind.Baseline.Test/JsonRpc/BaselineModuleTests.cs [hunk 1]
```diff
@@ -79,7 +80,7 @@ public async Task deploy_deploys_the_contract()
 
             await testRpc.AddBlock();
 
-            testRpc.BlockTree.Head.Number.Should().Be(5);
+            testRpc.BlockTree.Head.Number.Should().Be(2);
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Baseline.Test/JsonRpc/BaselineModuleTests.cs:83]: why would this number change from 5 to 2?

File: src/Nethermind/Nethermind.Core.Test/Blockchain/TestBlockchain.cs [hunk 2]
```diff
@@ -172,28 +172,39 @@ protected virtual async Task<TestBlockchain> Build(ISpecProvider specProvider =
...
-            // {
-            //     throw new InvalidOperationException("Failed to process genesis in 1s.");
-            // }
-
+            await WaitAsync(_resetEvent, "Failed to process genesis in time.");
             await AddBlocksOnStart();
             return this;
         }
 
+        private async Task WaitAsync(SemaphoreSlim semaphore, string error, int timeout = 1000)
```
Comment (tkstanczak) [inline src/Nethermind/Nethermind.Core.Test/Blockchain/TestBlockchain.cs:180]: we have a helper class for it right? AsyncWait or sth

File: src/Nethermind/Nethermind.Baseline.Test/JsonRpc/BaselineModuleTests.cs (hunk 1 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Baseline.Test/JsonRpc/BaselineModuleTests.cs:83]: I removed additional blocks deployed at start

File: src/Nethermind/Nethermind.Core.Test/Blockchain/TestBlockchain.cs (hunk 2 above)
Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Core.Test/Blockchain/TestBlockchain.cs:180]: its used here in 2nd overload, we have it for WaitHandle not for Semaphore

**Pull request merged**